graspologic/nominate/spectralVN.py

```diff
@@ -0,0 +1,268 @@
+        unique attributes (and therefore the shape along axis 0 of `unique_att_`) is equal to the number of seeds ( the
+        shape along axis 0 of `attr_labels_`).
+    distance_matrix_ : np.ndarray
+        The euclidean distance from each seed vertex to each vertex.
+        Shape is ``(number_vertices, number_unique_attributes)`` if attributed or Shape is
+        ``(number_vertices, number_seed_vertices)`` if unattributed.
+
+    References
+    ----------
+    .. [1] Fishkind, D. E.; Lyzinski, V.; Pao, H.; Chen, L.; Priebe, C. E. Vertex nomination schemes for membership
+        prediction. Ann. Appl. Stat. 9 2015. https://projecteuclid.org/euclid.aoas/1446488749
+
+    .. [2] Jordan Yoder, Li Chen, Henry Pao, Eric Bridgeford, Keith Levin, Donniell E. Fishkind, Carey Priebe,
+        Vince Lyzinski, Vertex nomination: The canonical sampling and the extended spectral nomination schemes,
+        Computational Statistics & Data Analysis, Volume 145, 2020.
+        http://www.sciencedirect.com/science/article/pii/S0167947320300074
+
+    """
+
+    def __init__(
+        self,
+        embedding: np.ndarray = None,
+        embeder: Union[str, BaseSpectralEmbed] = "ASE",
+        persistent: bool = True,
+    ):
+        super().__init__(multigraph=False)
+        self.embedding = embedding
+        if self.embedding is None or not persistent:
+            if isinstance(embeder, BaseSpectralEmbed):
+                self.embedder = embeder
+            elif embeder == "ASE":
+                self.embedder = ase()
+            elif embeder == "LSE":
+                self.embedder = lse()
+            else:
+                raise TypeError
+        elif np.ndim(embedding) != 2:
+            raise IndexError("embedding must have dimension 2")
+        self.persistent = persistent
+        self.attr_labels_ = None
+        self.unique_att_ = None
+        self.distance_matrix_ = None
+
+    @staticmethod
+    def _make_2d(arr: np.ndarray) -> np.ndarray:
+        # ensures arr is two or less dimensions.
+        # if 1d, adds unique at each index on
+        # the second dimension.
+        if not np.issubdtype(arr.dtype, np.integer):
+            raise TypeError("Argument must be of type int")
+        arr = np.array(arr, dtype=np.int)
+        if np.ndim(arr) > 2 or (arr.ndim == 2 and arr.shape[1] > 2):
+            raise IndexError("Argument must have shape (n) or (n, 1) or (n, 2).")
+        elif np.ndim(arr) == 1 or arr.shape[1] == 1:
+            arr = arr.reshape(-1, 1)
+            arr = np.concatenate((arr, np.arange(arr.shape[0]).reshape(-1, 1)), axis=1)
+        return arr
+
+    def _pairwise_dist(self, y: np.ndarray, metric: str = "euclidean") -> np.ndarray:
+        # wrapper for scipy's cdist function
+        # y should give indexes
+        y_vec = self.embedding[y[:, 0].astype(np.int)]
+        dist_mat = distance.cdist(self.embedding, y_vec, metric=metric)
+        return dist_mat
+
```

Comment:
what's the logic behind doing this as opposed to NearestNeighbors in sklearn (kd-tree)?

Reply:
Logic is that I need to to have access to and return the actual distance matrix, and (unless k is specified) the pairwise distance between each seed and every other vertex is used, in which case `cdist` is faster than `NearestNeighbors`.

Reply:
do u have a notebook showing this?

Reply:
reason is, if the difference is not significant, I think it would probably be better just to use nearest neighbors (takes up less space, more easily readable). If it is significantly faster, maybe this is something worth bringing up as an issue in sk-learn @bdpedigo ?

Reply:
switched to using nearest neighbors